### ISSUE TITLE
Make enum metada 

### ISSUE THREAD

**Issue by josueBarretogit** [lise-henry/epub-builder#48]
At the moment when calling `epub_builder.metada` it receives `key` which is `AsRef<str>`, it returns `Err` when you don´t pass a valid key, wouldn it be better to have an enum `Metada` so that you can´t pass an invalid key? btw I really love this crate you guys made it so simple to make epubs, I thought this was gonna be like using a pdf crate but this one turned out to be really simple to use.

**Comment by duyk471** [comment]
Sounds great, I've already added the PageDirection Enum for PageDirection:

```rust
impl ToString for PageDirection {
    fn to_string(&self) -> String {
        match &self {
            PageDirection::Rtl => "rtl".into(),
            PageDirection::Ltr => "ltr".into(),
        }
    }
}
```

This is only applied for values that have fixed options, such as Page Direction, EPUB_VERSION (I guess there are only two of them, maybe some other properties such as "primary-writing-mode" which is needed when generating Japanese and Chinese ebooks, but they're pretty niche so I guess they won't be added soon)

**Comment by crowdagger** [comment]
Yes it would be good, might look at it when I have  the energy :)
